app/src/main/java/com/testtask/apptesttask/model/repository/CharactersRepository.kt

```diff
@@ -46,32 +41,29 @@ class CharactersRepository @Inject constructor(
                             it.series,
                             favorites[id]?.favorite ?: false
                         )
-                        characters.add(character)
-                        character
                     }
                 }
             )
                     .subscribeOn(schedulers.io())
                     .observeOn(schedulers.ui())
 
-    fun favorCharacter(position: Int): Single<MutableList<Character>> {
-        val character = characters[position].copy(favorite = !characters[position].favorite)
-        val id = character.id
-        characters[position] = character
-
-        val favoriteCharacters = prefs.favoritesCharacters
+    fun favorCharacter(character: Character): Single<Character> =
 
-        if (favoriteCharacters.contains(id)) {
-            favoriteCharacters.remove(id)
-            prefs.favoritesCharacters = favoriteCharacters
-        } else {
-            favoriteCharacters[id] = character
-            prefs.favoritesCharacters = favoriteCharacters
-        }
+            Single.just(character)
+                    .map {
+                        val favoriteCharacter = it.copy(favorite = !it.favorite)
+                        val favoriteCharacters = prefs.favoritesCharacters
```

Comment:
Rename `favoriteCharacters` as `characters` and `favoriteCharacter` as `newCharacter`.